docker_mcp/core/migration/manager.py

```
@@ -0,0 +1,309 @@
+            self.logger.info(
+                "Force stopping containers",
+                stack=stack_name,
+                containers=running_containers,
+            )
+            
+            # Force stop each container
+            for container in running_containers:
+                stop_cmd = ssh_cmd + [f"docker kill {container}"]
+                await asyncio.get_event_loop().run_in_executor(
+                    None,
+                    lambda: subprocess.run(  # nosec B603
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)